src/services/importService.ts

```diff
@@ -0,0 +1,80 @@
+import { Collection } from 'types/collection';
+import { ElectronFile, FileWithCollection } from 'types/upload';
+import { runningInBrowser } from 'utils/common';
+import { getElectronFiles } from 'utils/upload';
+
+class ImportService {
+    ElectronAPIs: any;
+    private allElectronAPIsExist: boolean = false;
+
+    constructor() {
+        this.ElectronAPIs = runningInBrowser() && window['ElectronAPIs'];
+        this.allElectronAPIsExist = !!this.ElectronAPIs?.exists;
+    }
+
+    async setDoneUploadingFiles() {
+        if (this.allElectronAPIsExist) {
+            this.ElectronAPIs.setToUploadFiles(null, null, null, true);
+        }
+    }
+
+    async getIfToUploadFilesExists(): Promise<boolean> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.getIfToUploadFilesExists();
+        }
+    }
+
+    async getElectronFile(filePath: string): Promise<ElectronFile> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.getElectronFile(filePath);
+        }
+    }
+
+    async showUploadFilesDialog(): Promise<string[]> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.showUploadFilesDialog();
+        }
+    }
+
+    async showUploadDirsDialog(): Promise<string[]> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.showUploadDirsDialog();
+        }
+    }
+
+    async getToUploadFiles() {
+        if (this.allElectronAPIsExist) {
+            const { filesPaths, collectionName, collectionIDs } =
+                this.ElectronAPIs.getToUploadFiles();
+            const files = await getElectronFiles(filesPaths);
```

Comment:
I was thinking if we could just return ElectronFiles on `getToUploadFiles` and hence move this logic to the electron layer?